pkg/api/store/cluster/cluster_store.go

```diff
@@ -55,15 +57,16 @@ func (t *transformer) transposeGenericConfigToDynamicField(data map[string]inter
 
 		var driver *v3.KontainerDriver
 		driverName := data["genericEngineConfig"].(map[string]interface{})[clusterprovisioner.DriverNameField].(string)
+		// iterate over kontainer drivers to find the one that maps to the genericEngineConfig DriverName ("kd-**") -> "example"
 		for _, candidate := range drivers {
 			if driverName == candidate.Name {
 				driver = candidate
 				break
 			}
 		}
-
-		if driver == nil {
-			return nil, fmt.Errorf("got unknown driver from kontainer-engine: %v", data[clusterprovisioner.DriverNameField])
+		if driver == nil { // driver is nil if it is never assigned
```

Comment:
gernally, comments on same line of code gets really hard to read. put them on the line above